dsgrid/config/input_dataset_requirements.py

```diff
@@ -0,0 +1,39 @@
+"""Defines dataset dimension requirements for a project."""
+
+from pydantic import conlist, Field
+from typing_extensions import Annotated
+
+from dsgrid.config.project_config import RequiredDimensionsModel, InputDatasetModel
+from dsgrid.data_models import DSGBaseModel
+
+
+class InputDatasetDimensionRequirementsModel(DSGBaseModel):
+    """Defines dataset dimension requirements."""
+
+    dataset_id: str
+    required_dimensions: Annotated[
+        RequiredDimensionsModel,
+        Field(
+            title="required_dimensions",
+            description="Defines required record IDs that must exist for each dimension.",
+        ),
+    ]
+
+
+class InputDatasetDimensionRequirementsListModel(DSGBaseModel):
+    """Defines a list of dataset dimension requirements."""
+
+    datasets: Annotated[
```

Comment:
Is this name correct given description? It's list of dataset dimension requirements

Reply:
If this is confusing, I can change it. Each entry in this list is a set of requirements for a dataset. `datasets` is shorter than `dataset_dimension_requirements`. It's also like an overlay to this class: https://github.com/dsgrid/dsgrid/blob/main/dsgrid/config/project_config.py#L695, and I like the symmetry.

Reply:
I think the symmetry is good, but if the label is the same as the InputDatasetListModel, I would expect it to take those model/input rather than the RequirementsListModel here. So personally, I would prefer a more nuanced label.

Reply:
This parameter is user facing and so consensus would be good. @elainethale and @ashreeta can also weigh in.

I'll elaborate on the symmetry argument. In this scenario the user will most likely copy the existing dataset section of the project config into a new JSON file, remove unrelated fields, edit the requirements, and then upload it through this new CLI command. While calling the field `dataset_dimension_requirements` is more specific, I think that `datasets` is simpler, provides sufficient explanation, and is the same name that is used in the project config (allowing copy/paste) to work. The new blob would look something like this:
```
datasets: [
  dataset_id: 'decarb_2023_dgen',
  required_dimensions: {
    single_dimensional: {
      sector: {base: ['res', 'com']},
      metric: {base: ['electricity_dpv']}
    }
  }
]
```

Reply:
I also prefer the more accurate name,`dataset_dimension_requirements`. The user already has to modify what they've copy-pasted ...